### DISCUSSION TITLE
Error a la hora de cargar elementos js | dist/index.html

### DISCUSSION THREAD

**Discussion by ide45** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#574] [Q&A]
Quiero desplegar la práctica de la pastelería. Ya generé por medio del comando `npm run serve` lo necesario para pasar los archivos a github. Cuando abro `index.html` a cualquier navegador me sale la página en blanco y la consola arroja este error:

![image](https://user-images.githubusercontent.com/91546625/161365737-c555086c-85cd-4505-8a2e-61419b5a513c.png)

**Comment by doguedogue** [comment]
`npm run serve`            es para levantar el servidor localmente (se ejecuta la página en http://localhost:8080/)

`npm install`                 para cargar los modulos (prueba este comando cuando descargues un proyecto para que te cargue los modulos)

`npm run build`           es para construir la distribución y publicar una página en un servidor  (te crea la carpeta /dist)

**Reply by ide45** [reply]
Perdón, me equivoqué a la hora de escribir el comando en la discusión. Ya ejecuté el comando `npm run build` se generó la carpeta dist. 

Cuando abro el index.html creado, me salen esos dos errores.

**Reply by carlos-gmx3** [reply]
No tienes que abrir index.html ni ejecutar Live serve
copia esta dirección en tu navegador http://localhost:8080/ despues de `npm run serve` y listo

**Reply by ide45** [reply]
El serve me va bien, lo que no funciona son los archivos de distribución, algo falla con la declaración de rutas que no permite que los scripts se carguen correctamente

**Comment by JonathanGuerraGro** [comment]
Hola, la verdad yo no se usar Vue, pero investigue un poco y solo he encontrado 2 cosas muy comunes en ese tipo de errores (de hecho tuve un problema similar en una practica anterior y era por las rutas).

Lo que encontre fue:

1. Si tienes un navegador con algun AdBlock podria estar bloqueandolo, podrias probar desactivando el AdBlock

2. Hay algun detalle por ahi con las rutas, por ahi encontre un reddit a alguien con un problema muy parecido, pagina en blanco y error de source, te dejo el link esperando que te sirva de algo.
https://www.reddit.com/r/learnprogramming/comments/8rd5y3/loading_failed_for_the_script_with_source/e0qdfak/

**Comment by Lfer1111** [comment]
@doguedogue una duda a ver si me pudieras ayudar.. ya tengo lista la pasteleria con vue.. incluso ya la subi a github en un repo, pero no he podido desplegar la pagina en GH-pages .. es un poco diferente a las anteriores practicas porque este es todo un proyecto construido con este framework.. aqui solo me aparece desplegable el readme, pero no entra al website en si como quisiera.. 
entonces la duda es como puedo hacerlo desplegable en github pages?

REPO
https://github.com/Lfer1111/Entrega5-vue-pastel

"PAGINA DESPLEGADA"
https://lfer1111.github.io/Entrega5-vue-pastel/


gracias de antemano

**Comment by doguedogue** [comment]
Asi es, es diferente, la verdad yo no pude con GH pages (hay que hacerle varias modificaciones o subir un archivo de configuración yaml), hay otras **tres** formas de hacerlo (o las que me se)

1. Subirlo a azure páginas estáticas, como nos enseñó el MC (está en el penúltimo video
2. Hacer un deploy de tu proyecto en VS Code:
      npm run build
      esto te genera la carpeta DIST
      Lo que está en DIST ya es algo que se puede publicar, es decir, lo que deberia hacer github pages es tomar esa carpeta  
      Lo puedes poner en tu carpeta, asi literal copiarlo y pegarlo en https://lfer1111.github.io/Entrega5-vue-pastel/
3. El resultado de DIST subirlo a netlify o el webhosting de tu preferencia

Yo lo hice con azure páginas estáticas y netlify

-     https://github.com/doguedogue/xocolatl-vue
-     https://icy-hill-0629dae0f.1.azurestaticapps.net/#/

**Comment by Lfer1111** [comment]
Muchas gracias!! Lo intentaré así como dices =)
